src/Database/Drivers/PgSqlDriver.php

```diff
@@ -256,7 +257,18 @@ public function getForeignKeys($table)
 				co.contype = 'f'
 				AND cl.oid = {$this->connection->quote($this->delimiteFQN($table))}::regclass
 				AND nf.nspname = ANY (pg_catalog.current_schemas(FALSE))
-		")->fetchAll();
+		";
+
+		$keys = [];
+		foreach ($this->connection->query($query) as $row) {
+			if (preg_match('~FOREIGN KEY\s*\((.+)\)\s*REFERENCES (.+)\((.+)\)(.*)$~iA', $row['definition'], $match)) {
```

Comment:
This is not systematic for PostgreSQL. There are columns for this purpose `confupdtype` and `confdeltype` (http://www.postgresql.org/docs/8.4/static/catalog-pg-constraint.html).

Reflection is, IMHO, internal thing and FK actions are not needed in Nette Database.